src/ts/forwarders/forwarder.ts

```diff
@@ -43,20 +43,26 @@ export interface DeploymentParams {
 }
 
 export interface RelayHubCallData {
-  to: string;
+  to?: string;
   data: string;
 }
 
-export interface ProxyAccountCallData extends RelayHubCallData {
+export interface ProxyAccountCallData {
+  to?: string;
   value?: BigNumberish;
+  data?: string;
 }
 
+type RequiredPick<T, TRequired extends keyof T> = T &
+  Pick<Required<T>, TRequired>;
+export type RequiredTo<T extends { to?: string }> = RequiredPick<T, "to">;
+
 /**
  * Provides common functionality for the RelayHub and the ProxyAccounts.
  * Possible to extend it with additional functionality if another
  * msg.sender solution emerges.
  */
-export abstract class Forwarder<T> {
+export abstract class Forwarder<TParams extends Partial<MinimalTx>> {
```

Comment:
what is motivation of this compared to just T?

Reply:
This allows me to know that T has the nullable to and data on it - so that I can check for them in the signAndEncode function